entrypoint.sh

```diff
@@ -100,37 +99,131 @@ function super_lint() {
   /action/lib/linter.sh
 }
 
+function get_files_matching_filetype() {
+  local matching_files=()
+  local filenames=("${@:2}")
+  local filetype=$1
+
+  for file in "${filenames[@]}"; do
+    filename=$(basename "$file")
+
+    if [[ "$filename" == *"$filetype" ]]; then
+      matching_files+=("$file")
+    fi
+  done
+  echo "${matching_files[@]}"
+}
+
+function linter_failed() {
+  return=$1
+
+  if [[ "${return:-1}" != 0 ]]; then
+    linter_exit_codes+=(["${linter[name]}"]="${return}")
+  else
+    return "${return}"
+  fi
+}
+
+function lint_files() {
+  exec &>> "${linter[logfile]}"
```

Comment:
This function takes arguments from the linter array, dynamically builds the command, evaluates the command, and passes the exit code to the linter_failed function for logging. It takes into account three conditions: 1) a linter is internally handling all file types to determine which ones to lint indicated by 'filetype=all' in linters.txt; 2) a linter is being run on specific files with, what I'm calling, an executor (i.e. npx--I am completely open to changing this terminology); and 3) the linter is being run on specific files as a standalone command (no executor).

Reply:
Executor is fine.

I'm not a huge fan of the `exec` approach on this first line though, changing the stdout and stderr of the current runtime entirely to the log file for the provided linter.  If anything else ever happens in between as a part of the goat, it would get sucked into that log file.

Instead I recommend we just redirect the `eval`s